WooCommerce/Classes/ViewRelated/JetpackSetup/AdminRole/AdminRoleRequiredViewModel.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import Yosemite
+
+/// View model for `AdminRoleRequiredView`
+///
+final class AdminRoleRequiredViewModel {
+    let username: String
+    let roleName: String
+
+    private let siteID: Int64
+
+    private let stores: StoresManager
+    init(siteID: Int64, stores: StoresManager = ServiceLocator.stores) {
+        self.siteID = siteID
+        self.stores = stores
+        self.username = stores.sessionManager.defaultCredentials?.username ?? ""
+        self.roleName = stores.sessionManager.defaultRoles
+            .map { $0.displayString() }
+            .joined(separator: ", ")
+    }
+
+    @MainActor
+    func reloadRoles() async throws -> Bool {
+        try await withCheckedThrowingContinuation { continuation in
+            let action = UserAction.retrieveUser(siteID: siteID) { [weak self] result in
+                guard let self = self else { return }
```

Comment:
❓  Should we resume the `continuation` by throwing an error when `self` is unavailable?

Reply:
I switched to using `weak self` in 54727fc4cee7411d31070e799dcbc00e49d98b83 since it's not straightforward to decide which error to throw here.